impl/src/main/java/org/jboss/forge/roaster/model/impl/AbstractJavaSourceMemberHolder.java

```diff
@@ -457,6 +458,14 @@ public O implementInterface(Class<?> type)
    public O implementInterface(JavaInterface<?> type)
    {
       O obj = addInterface(type);
+
+      if (type instanceof JavaInterfaceSource) {
```

Comment:
Maybe the method parameter type could be changed to JavaInterfaceSource? What could that break?